### DISCUSSION TITLE
Help resolve: I am unable to push nextJS project to github?

### DISCUSSION THREAD

**Discussion by ChrisB007** [vercel/next.js#29233] [Help]
Working on a project that I have been pushing to git, but my latest attempt seems impossible for some reason. Here is the error message that I am getting:

``` 
remote: warning: File .next/cache/webpack/client-development/5.pack is 62.92 MB; this is larger than GitHub's recommended maximum file size of 50.00 MB
remote: error: Trace: 2de90a801f920bc9f9b8e66d5ca6fddd441e7d126991facb3138b1bb8cc03900
remote: error: See http://git.io/iEPt8g for more information.
remote: error: File .next/cache/webpack/client-development/16.pack is 117.14 MB; this exceeds GitHub's file size limit of 100.00 MB
remote: error: File .next/cache/webpack/client-development/16.pack is 117.15 MB; this exceeds GitHub's file size limit of 100.00 MB
remote: error: GH001: Large files detected. You may want to try Git Large File Storage - https://git-lfs.github.com.
To https://github.com/ChrisB007/nextFinalPublicTrades.git
 ! [remote rejected] main -> main (pre-receive hook declined)
error: failed to push some refs to 'https://github.com/ChrisB007/nextFinalPublicTrades.git'

```

What I tried doing in order to resolve the issue

```
Step 0: Followed steps to remove large files as provided by git docs [https://docs.github.com/en/repositories/working-with-files/managing-large-files/about-large-files-on-github#removing-files-from-a-repositorys-history])
Step 1: I deleted the .next folder
Step 2: I committed the change
Step 3: Added .next to .gitignore file
Step 4: Pushed the change.

```

The issue still persists and getting the same error after that. Please help

**Comment by ijjk** [comment]
Hi, it sounds like the files in the `.next` folder are still in the git history see here for info are removing them from the git history https://docs.github.com/en/repositories/working-with-files/managing-large-files/about-large-files-on-github#removing-files-from-a-repositorys-history

**Reply by ChrisB007** [reply]
Thanks for your response, I actually did try that already but didnt work

**Reply by ijjk** [reply]
So it still fails to push after removing the `.next` files from git history and no entries are shown with `git log --stat --all -- .next`?

**Reply by anthony-effectual** [reply]
I'm having the same issue.

I tried  `git rm --cached large_file` and it returned "did not match any file"

**Reply by BarkMulcher** [reply]
@ChrisB007 could you try adding "/.next" in your .gitignore rather than ".next"?

**Reply by barayiti01** [reply]
I am experiencing a similar issue and I've deleted the .next and node_modules folders. I also tried both alternatives of adding "./next or .next" in the .gitignore file but no luck

**Comment by c2bsolutions** [comment]
Hello Chris,

I am experiencing the same error. I have tried ignoring and deleting the files and using git-lfs to track the files to avoid the error, but all my attempts have been unsuccessful. Were you able to find a solution for this?

Thanks,

**Comment by Ferdinand1945** [comment]
did someone solved this one? I am having the same issue

Thanks!

**Comment by BarkMulcher** [comment]
Hi, I had the same issue, but in my .gitignore, I added "/.next", and that allowed me to push with not even a warning. You may have tried to ignore a ".next" ***file*** that does not exist. Rather the /.next will ignore a ***folder***.

**Reply by GabenGar** [reply]
Well not really, `/.next` will ignore whatever entry with the name `.next` at the root of the git project, while `.next` ignores any entry with `.next` in any subfolder within the git project.
I suspect there is some faulty example nextjs app which excludes `.next` from ignore for some reason.

**Reply by saeidniyabati2** [reply]
thank you .it worked for me

**Reply by NafiurSiddiqui** [reply]
> thank you .it worked for me

What did you do eventually?

**Reply by bunnyBites** [reply]
This also helped me, as the issue is because of file size (commit)

**Comment by NafiurSiddiqui** [comment]
Hi, I have tried everything mentioned here. none of them worked. Any way to get out of this?

**Reply by NafiurSiddiqui** [reply]
> Hi, I have tried everything mentioned here. none of them worked. Any way to get out of this?

Ok, I am not having this problem when I install with ` npx create-next-app@latest`. This issue persists with `yarn` for me.

**Reply by josealvarez97** [reply]
@NafiurSiddiqui Did you find a solution for yarn?

**Reply by NafiurSiddiqui** [reply]
> @NafiurSiddiqui Did you find a solution for yarn?

Unfortunately no. I had limited time to look further into it. Traded off with `npm` . Plz, do post it here if you found a solution for yarn.

**Reply by josealvarez97** [reply]
@NafiurSiddiqui Thank you for the prompt response. In my case, it seems that the issue was that GitHub was not working well this morning for many people. See below for secondary and official sources.

> In my case, it seems that the issue was that GitHub was down today (Mar 27th, 2023) and everyone was having the same error message. I confirmed this by testing in multiple repos of different organizations. It resolved itself after approximately 1 hour.
> 
> See here for secondary sources:
> 
> https://news.ycombinator.com/item?id=35325709
> 
> https://www.githubstatus.com/incidents/52z0j6phhnjs
> 
> ![image](https://user-images.githubusercontent.com/21250471/227950557-ccee8159-8d20-4bb2-8cae-fb5df4b6f2a9.png)
> 
> ![image](https://user-images.githubusercontent.com/21250471/227951056-a90203e8-7062-4634-a318-137b5b02c092.png)

**Comment by josealvarez97** [comment]
It is very frustrating that this issue arises days after integrating Vercel's cloud services with our repository. Can this be addressed?

**Reply by josealvarez97** [reply]
In my case, it seems that the issue was that GitHub was down today (Mar 27th, 2023) and everyone was having the same error message. I confirmed this by testing in multiple repos of different organizations. It resolved itself after approximately 1 hour.

See here for secondary sources:

https://news.ycombinator.com/item?id=35325709

https://www.githubstatus.com/incidents/52z0j6phhnjs

![image](https://user-images.githubusercontent.com/21250471/227950557-ccee8159-8d20-4bb2-8cae-fb5df4b6f2a9.png)

![image](https://user-images.githubusercontent.com/21250471/227951056-a90203e8-7062-4634-a318-137b5b02c092.png)

**Reply by NafiurSiddiqui** [reply]
Thank you for the update. Hopefully I won't be facing this exciting errors anymore next time with yarn. My issue was not with GH being down but the whole `large file cache` issue. Glad it worked out for you. 🙌

**Comment by atalayio** [comment]
Hi your problem is it still same? Here quick solution for you it worked for me!

First of all you can use the following decision to cancel unsaved restrictions:

git reset --hard HEAD

and try this:

git filter-branch --force --index-filter "git rm --cached --ignore-unmatch '.next/cache/webpack/client-development/11.pack'" --prune-empty --tag-name-filter cat -- --all
(I got file size error for "11.pack", if you have different file just change it to path.)

last one:

git push --force

It worked fine.

**Reply by RodrLucas** [reply]
Thanks!! It worked for me!!

**Reply by donprasetiyo** [reply]
Thanks. This works.

**Reply by AbirHal** [reply]
thanks ,it worked !!

**Reply by Ghufran496** [reply]
it worked. Thanks

**Reply by bunnyBites** [reply]
It worked... at last. Thanks

**Reply by atalayio** [reply]
Can you mark my answer as solved, by the way I'm happy you solved it too guys! @ChrisB007

**Reply by mahautlatinis** [reply]
Hi ! I have the same issue but it's not solved on my side with these commands. 
I was working on new branches and I can't push it 😭, would you have some suggestions please ?

**Reply by abenomran** [reply]
Thanks!!!

**Comment by WANHO-O** [comment]
I succeeded using this code
`git filter-branch --force --index-filter "git rm -rf --cached --ignore-unmatch .next" --prune-empty --tag-name-filter cat -- --all
`

**Reply by mosharafali** [reply]
Thanks
